src/libs/actions/Session/index.js

```diff
@@ -78,10 +82,36 @@ function signOut() {
     Timing.clearData();
 }
 
-function signOutAndRedirectToSignIn() {
+/**
+ * @param {String} [authTokenType] The type of auth token to check
+ */
+function signOutAndRedirectToSignIn(authTokenType = '') {
     signOut();
     redirectToSignIn();
     Log.info('Redirecting to Sign In because signOut() was called');
+    if (SessionUtils.isAnonymousUser(authTokenType)) {
+        Linking.getInitialURL().then((url) => {
+            const reportID = ReportUtils.getReportIDFromLink(url);
+            if (reportID) {
+                Report.setLastOpenedPublicRoom(reportID);
+            }
+        });
+    }
+}
+
+/**
+ *
+ * @param {String} authTokenType The type of auth token to check
+ * @param {Function} callback The callback to execute if the action is allowed
+ * @returns {*} callback's result
+ */
+function checkIfActionIsAllowed(authTokenType, callback) {
+    return () => {
+        if (SessionUtils.isAnonymousUser(authTokenType)) {
+            return signOutAndRedirectToSignIn(authTokenType);
+        }
+        return callback();
+    };
```

Comment:
Not tested but I believe this will cause regression when callback requires params. Also should have null safety check.
And there might be some codes which check callback existence. i.e. `if (!onPress)`
```suggestion
    if (typeof callback !== 'function') return callback;
    return (...args) => {
        if (SessionUtils.isAnonymousUser(authTokenType)) {
            return signOutAndRedirectToSignIn(authTokenType);
        }
        return callback(...args);
```

Reply:
Actually, I think this is optimized code:
```suggestion
    if (SessionUtils.isAnonymousUser(authTokenType)) {
        return () => signOutAndRedirectToSignIn(authTokenType);
    }
    return callback;
```

Reply:
Oh nice catch! Thanks also for this suggestion, updating it!